db/blip_handler.go

```diff
@@ -1060,48 +1076,66 @@ func (bh *blipHandler) downloadOrVerifyAttachments(sender *blip.Sender, body Bod
 			}
 
 			return nil, proveAttErr
-		})
+		}, docID)
 }
 
 func (bsc *BlipSyncContext) incrementSerialNumber() uint64 {
 	return atomic.AddUint64(&bsc.handlerSerialNumber, 1)
 }
 
-func (bsc *BlipSyncContext) addAllowedAttachments(attDigests []string) {
-	if len(attDigests) == 0 {
+func (bsc *BlipSyncContext) addAllowedAttachments(docID string, attMeta []AttachmentStorageMeta) {
+	if len(attMeta) == 0 {
 		return
 	}
 
 	bsc.lock.Lock()
 	defer bsc.lock.Unlock()
 
 	if bsc.allowedAttachments == nil {
-		bsc.allowedAttachments = make(map[string]int, 100)
+		bsc.allowedAttachments = make(map[string]AllowedAttachment, 100)
 	}
-	for _, digest := range attDigests {
-		bsc.allowedAttachments[digest] = bsc.allowedAttachments[digest] + 1
+	for _, attachment := range attMeta {
+		if bsc.blipContext.ActiveProtocol() == BlipCBMobileReplicationV3 {
+			key := docID + attachment.digest
+			if _, found := bsc.allowedAttachments[key]; !found {
+				bsc.allowedAttachments[key] = AllowedAttachment{counter: 1}
+			}
+		} else {
+			att := bsc.allowedAttachments[attachment.digest]
+			att.counter = att.counter + 1
+			att.docID = docID
+			att.version = attachment.version
```

Comment:
If att already exists in bsc.allowedAttachments, is there a need to update docID/version?